impl/src/main/java/org/jboss/forge/roaster/spi/JavaParserImpl.java

```diff
@@ -55,13 +57,28 @@
       return parseUnit(data).getGoverningType();
    }
 
+   private Charset getDefaultEncoding()
+   {
+      return Roaster.getDefaultEncoding();
+   }
+
    @Override
    public JavaUnit parseUnit(InputStream data)
    {
+      return parseUnit(data, getDefaultEncoding());
+   }
+
+   @Override
+   public JavaUnit parseUnit(InputStream data, Charset encodingIfText)
+   {
+      return parseUnit(data, encodingIfText.name());
+   }
+
+   private JavaUnit parseUnit(InputStream data, String encodingIfText)
+   {
       try
       {
-         String encoding = System.getProperty("file.encoding", "ISO8859_1");
-         char[] source = Util.getInputStreamAsCharArray(data, data.available(), encoding);
+         char[] source = Util.getInputStreamAsCharArray(data, -1, encodingIfText);
```

Comment:
See the javadoc of Util#getInputStreamAsCharArray(InputStream stream, int length, String encoding) throws IOException.
[javadoc [Returns the given input stream's contents as a character array. If a length is specified (i.e. if length != -1), this represents the number of bytes in the stream. Note this doesn't close the stream.]]

This '-1' is preferable because the #available() is a non-blocking readable size of stream, so it is not correct in some cases (see InputStream#available()).